frontend/packages/integration-tests-cypress/tests/crud/namespace-crud.spec.ts

```diff
@@ -53,54 +54,113 @@ describe('Namespace', () => {
+    listPage.rows.shouldBeLoaded();
     projectDropdown.selectProject('All Projects');
     projectDropdown.shouldContain('All Projects');
     cy.log(
       'Nav from list page to details page should change Project from "All Projects" to resource specific project',
     );
-    cy.get(`[data-test-rows="resource-row"]`)
-      .first()
-      .find('a')
-      .first()
-      .click();
+    listPage.rows.clickFirstLinkInFirstRow();
+    detailsPage.isLoaded();
     projectDropdown.shouldNotContain('All Projects'); // after drilldown to Details page, project should be specific to resource
     cy.log(
       'Nav back to list page from details page via sidebar nav menu should change Project back to "All Projects"',
     );
     nav.sidenav.clickNavLink(['Workloads', 'Secrets']);
+    listPage.rows.shouldBeLoaded();
     projectDropdown.shouldContain('All Projects');
     cy.log(
       'Nav back to list page from details page via breadcrumb should change Project back to "All Projects"',
     );
-    cy.get(`[data-test-rows="resource-row"]`)
-      .first()
-      .find('a')
-      .first()
-      .click();
+    listPage.rows.clickFirstLinkInFirstRow();
+    detailsPage.isLoaded();
     projectDropdown.shouldNotContain('All Projects'); // after drilldown to Details page, project should be specific to resource
-    cy.byLegacyTestID('breadcrumb-link-0').click();
+    detailsPage
+      .breadcrumb(0)
+      .contains('Secrets')
+      .click();
+    listPage.rows.shouldBeLoaded();
     projectDropdown.shouldContain('All Projects');
   });
 
   it('nav and breadcrumbs restores last selected Project when navigating from details to list view', () => {
     nav.sidenav.clickNavLink(['Workloads', 'Secrets']);
     projectDropdown.selectProject('default');
     projectDropdown.shouldContain('default');
-    cy.get(`[data-test-rows="resource-row"]`)
-      .first()
-      .find('a')
-      .first()
-      .click();
+    listPage.rows.clickFirstLinkInFirstRow();
+    detailsPage.isLoaded();
     projectDropdown.shouldContain('default');
     nav.sidenav.clickNavLink(['Workloads', 'Secrets']);
     projectDropdown.shouldContain('default');
-    cy.get(`[data-test-rows="resource-row"]`)
-      .first()
-      .find('a')
-      .first()
-      .click();
+    listPage.rows.clickFirstLinkInFirstRow();
+    detailsPage.isLoaded();
     projectDropdown.shouldContain('default');
-    cy.byLegacyTestID('breadcrumb-link-0').click();
+    detailsPage
+      .breadcrumb(0)
+      .contains('Secrets')
+      .click();
+    listPage.rows.shouldBeLoaded();
     projectDropdown.shouldContain('default');
   });
+
+  ['Roles', 'RoleBindings'].forEach((rolesOrBindings) => {
+    it(`tests ${rolesOrBindings} and Cluster${rolesOrBindings} detail page breadcrumbs`, () => {
+      nav.sidenav.clickNavLink(['User Management', rolesOrBindings]);
+
+      cy.log(`test ${rolesOrBindings} & All Projects`);
+      projectDropdown.selectProject('All Projects');
```

Comment:
"All Namespaces" if running against native k8s.

Reply:
Due to limitations we discussed offline, I have filed a BZ to track this issue: [Cypress: Implement method to determine whether running OpenShift or K8s native](https://bugzilla.redhat.com/show_bug.cgi?id=1930964)